### PR TITLE
Merged jsoup scrappers with backend

### LINKED ISSUE #66: Merge new Jsoup scrappers with back-end
- [x] back-end should work with new scrappers
- [x] tests should pass

### PR DESCRIPTION
**Pull request by mr-tompkins**

### CODE REVIEW

File: src/main/java/robot/bonito/BonitoService.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        } catch (Exception e) {
+            //TODO log exception
+            System.err.println("An error occurred in a scrapper. No books have been updated.");
+            throw new ScrappingException("An error occurred in a scrapper. No books have been updated.", e);
+        }
+        //TODO add some mechanism to distinguish if a book is already in the repo.
+        bonitoBookRepository.deleteAll();
+        bonitoBookRepository.saveAll(BonitoBookMapper.toBonitoBook(books));
+    }
+}
```
Comment (KeshwaY) [inline src/main/java/robot/bonito/BonitoService.java:51]:
```suggestion
}

```

File: src/main/java/robot/BookProvider.java [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+/**
+ * @author Dominik Żebracki
+ */
+public interface BookProvider {
+
+    Books provideBooks();
+    boolean updateBooks();
+    Books updateAndProvideBooks();
+}
```
Comment (KeshwaY) [inline src/main/java/robot/BookProvider.java:11]:
```suggestion
}

```

File: pom.xml [hunk 3]
```diff
@@ -152,6 +163,16 @@
...
+                        <platforms>
+                            <platform>
+                                <architecture>arm64</architecture>
+                                <os>linux</os>
+                            </platform>
+                            <platform>
+                                <architecture>amd64</architecture>
+                                <os>linux</os>
+                            </platform>
+                        </platforms>
```
Comment (KeshwaY) [inline pom.xml:175]: This should look like in current `main` branch.

**Review by KeshwaY — DISMISSED** [review]

**Review by deetree — APPROVED** [review]
ok

File: src/main/java/robot/bonito/BonitoService.java (hunk 1 above)
Comment (deetree) [reply src/main/java/robot/bonito/BonitoService.java]: outdated

File: src/main/java/robot/BookProvider.java (hunk 2 above)
Comment (deetree) [reply src/main/java/robot/BookProvider.java]: outdated

**Pull request merged**